pyradiosky/skymodel.py

```diff
@@ -2623,15 +2652,143 @@ def select(
+                    np.nonzero(
+                        (skyobj.lon[component_inds] >= lon_range[0])
+                        & (skyobj.lon[component_inds] <= lon_range[1])
+                    )[0]
+                ]
+
+        if brightness_freq_range is not None:
+            if not np.atleast_1d(brightness_freq_range).size == 2:
+                raise ValueError("brightness_freq_range must have 2 elements.")
+
+        if min_brightness is not None or max_brightness is not None:
+            if skyobj.spectral_type == "spectral_index":
+                raise NotImplementedError(
+                    "Flux cuts with spectral index type objects is not supported yet."
+                )
+            freq_inds_use = None
+
+            if skyobj.freq_array is not None:
+                if brightness_freq_range is not None:
+                    freq_inds_use = np.where(
+                        (skyobj.freq_array >= np.min(brightness_freq_range))
+                        & (skyobj.freq_array <= np.max(brightness_freq_range))
+                    )[0]
+                    if freq_inds_use.size == 0:
+                        raise ValueError(
+                            "No object frequencies in specified range for flux cuts."
+                        )
+                else:
+                    freq_inds_use = np.arange(skyobj.Nfreqs)
+            if min_brightness is not None:
+                if not isinstance(
+                    min_brightness, Quantity
+                ) or not min_brightness.unit.is_equivalent(self.stokes.unit):
+                    raise TypeError(
+                        "min_brightness must be a Quantity object with units that can "
+                        f"be converted to {self.stokes.unit}"
+                    )
+                min_brightness = min_brightness.to(skyobj.stokes.unit)
+                if freq_inds_use is None:
+                    # written this way to avoid multi-advanced indexing
+                    stokes_use = skyobj.stokes[0][:, component_inds]
+                    assert stokes_use.shape == (skyobj.Nfreqs, component_inds.size)
+                else:
+                    # written this way to avoid multi-advanced indexing
+                    stokes_use = skyobj.stokes[0][freq_inds_use, :]
+                    stokes_use = stokes_use[:, component_inds]
+                    assert stokes_use.shape == (
+                        freq_inds_use.size,
+                        component_inds.size,
+                    )
+
+                component_inds = component_inds[
+                    np.nonzero(
+                        np.min(stokes_use.value, axis=0) >= min_brightness.value
+                    )[0]
+                ]
+
+            if max_brightness is not None:
+                if not isinstance(
+                    max_brightness, Quantity
+                ) or not max_brightness.unit.is_equivalent(self.stokes.unit):
+                    raise TypeError(
```

Comment:
Isn't this possible via the decorator to the method?

Reply:
No it's not because it requires knowledge on the object (self.stokes.unit) which is apparently not accessible to the decorator (I tried that first).

Reply:
Ah, got it, should have noticed that.